dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/impl/K8sTaskExecutor.java

```diff
@@ -184,8 +187,11 @@ public Job buildK8sJob(K8sTaskMainParameters k8STaskMainParameters) {
                 .endTemplate()
                 .withBackoffLimit(retryNum)
                 .endSpec();
-
-        return jobBuilder.build();
+        job = jobBuilder.build();
+        if (!StringUtils.isEmpty(pullSecret)) {
+            job.getSpec().getTemplate().getSpec()
+                    .setImagePullSecrets(singletonList(new LocalObjectReference(pullSecret)));
+        }
```

Comment:
I'm a bit confused about this code. Why not use `withImagePullSecrets`?

Reply:
Looks like we can use this usage avoid empty case
  ```
   .withCommand(commands.size() == 0 ? null : commands)
                .withArgs(args.size() == 0 ? null : args)
  ```

Reply:
done